content/docs/guides/auth.md

```diff
@@ -421,6 +421,76 @@ channel = google_auth_transport_grpc.secure_authorized_channel(
 stub = helloworld_pb2.GreeterStub(channel)
 ```
 
+##### With server authentication SSL/TLS and a custom header with token
+
+Client:
+
+```python
+import grpc
+import helloworld_pb2
+
+class GrpcAuth(grpc.AuthMetadataPlugin):
+    def __init__(self, key):
+        self._key = key
+
+    def __call__(self, context, callback):
+        callback((('rpc-auth-header', self._key),), None)
+
+with open('path/to/root-cert', 'rb') as fh:
+    root_cert = fh.read()
+
+channel = grpc.secure_channel(
+    'myservice.example.com:443',
+    grpc.composite_channel_credentials(
+        grpc.ssl_channel_credentials(root_cert),
+        grpc.metadata_call_credentials(
+            GrpcAuth('access_key')
+        )
+    )
+)
+
+stub = helloworld_pb2.GreeterStub(channel)
+```
+
+Server:
+
+```python
+import grpc
+import helloworld_pb2
+from concurrent import futures
+
+class AuthInterceptor(grpc.ServerInterceptor):
+    def __init__(self, key):
+        self._valid_metadata = ('rpc-auth-header', key)
+
+        def deny(_, context):
+            context.abort(grpc.StatusCode.UNAUTHENTICATED, 'Invalid key')
+
+        self._deny = grpc.unary_unary_rpc_method_handler(deny)
+
+    def intercept_service(self, continuation, handler_call_details):
+        if (
+            handler_call_details.invocation_metadata
+            and handler_call_details.invocation_metadata[0] == self._valid_metadata
+        ):
+            return continuation(handler_call_details)
```

Comment:
Nit: An explicit `else` would be nice here to aid readability, especially considering the indentation.

Reply:
@bartolootrit - can you make this update?

Reply:
Done

Reply:
Thanks!